### ISSUE TITLE
predict_parts() at dataset level

### ISSUE THREAD

**Issue by Tato14** [ModelOriented/DALEX#554]
Thanks for the great tool!

I was wondering if there is any way to perform `predict_parts()` at dataset level instead of the instance level, hence accepting more than one line in the `new_observation` param. I guess this would be more similar to `model_parts()` since the later evaluates explainability at the dataset level.

**Comment by Tato14** [comment]
Then, how I group the different contributions between models?  I just want to end up with an output similar to `model_parts()`. Should I use `mean()` for every variable?

**Comment by hbaniecki** [comment]
SHAP framework proposes to calculate the mean of absolute contribution values.

Note that feature attributions and permutational feature importance (default `model_parts`) have different interpretations, best explained in https://mindfulmodeler.substack.com/p/shap-is-not-all-you-need

![image](https://user-images.githubusercontent.com/32574004/235915118-00154feb-320c-45ae-847d-7a01837d2253.png)

**Comment by Tato14** [comment]
Thanks for the great resource. However, I am unsure of what you mean. Is because `model_parts()` does not use SHAP as a aggregator? And perform `mean()` on `predict_parts()` would be more similar to SHAP? If so, is it possible to aggregate the meaures in the same way that `model_parts()` do?

**Comment by hbaniecki** [comment]
>  Is because model_parts() does not use SHAP as a aggregator? And perform mean() on predict_parts() would be more similar to SHAP?

Yes, `model_parts(type="variable_importance")` calculates permutation-based feature importance as described in https://ema.drwhy.ai/featureImportance.html

In Python, `model_parts(type="shap_wrapper")` is available to calculate SHAP importance with the `shap` package.

> If so, is it possible to aggregate the measures in the same way that model_parts() do?

"the measures" meaning? `model_parts` measures performance loss after permutation, while `predict_parts` estimates feature attributions.

**Comment by hbaniecki** [comment]
assuming this is fixed
